packages/nix/packages/pf-eks-resume.sh

```diff
@@ -0,0 +1,286 @@
+  subnet_ids=$(aws eks describe-cluster --name "$EKS_CLUSTER_NAME" --query "cluster.resourcesVpcConfig.subnetIds[]" --output text)
+
+  if [ -z "$subnet_ids" ]; then
+    echo "Failed to get subnet IDs for the EKS cluster." >&2
+    return 1
+  fi
+
+  # Convert space-separated subnet IDs to a JSON array
+  subnet_ids_json=$(echo "$subnet_ids" | tr '\t' '\n' | jq -R . | jq -s .)
+
+  # Step 2: Query ASGs and filter by prefix and subnet IDs
+  asgs=$(aws autoscaling describe-auto-scaling-groups --query "AutoScalingGroups[?starts_with(AutoScalingGroupName, '${prefix}')] | [].[AutoScalingGroupName, VPCZoneIdentifier]" --output json)
+
+  echo "$asgs" | jq -c '.[]' | while IFS= read -r item; do
+    # Extract the nat and subnet values
+    nat=$(echo "$item" | jq -r '.[0]')
+    subnet=$(echo "$item" | jq -r '.[1]')
+
+    # Check if the subnet is in the list of subnets
+    if echo "$subnet_ids_json" | jq -e --arg subnet "$subnet" 'index($subnet) | . != null' >/dev/null; then
+      if aws autoscaling update-auto-scaling-group --auto-scaling-group-name "$nat" --min-size "$min_size" --desired-capacity "$desired_capacity"; then
+        echo "Successfully updated $nat to min-size $min_size and desired-capacity $desired_capacity." >&2
+      else
+        echo "Failed to update $asg." >&2
+      fi
+    else
+      echo "Subnet $subnet is not in the list. Skipping NAT $nat." >&2
+    fi
+  done
+}
+
+scale_asg_controller_nodes_up() {
+  local nodegroup_prefix="controllers-"
+
+  # Fetch all ASGs for the given EKS cluster and nodegroup prefix
+  asgs=$(aws autoscaling describe-auto-scaling-groups | jq -c ".AutoScalingGroups[] | select(
+             (.Tags[] | select(.Key == \"eks:cluster-name\" and .Value == \"$EKS_CLUSTER_NAME\")) and
+             (.Tags[] | select(.Key == \"eks:nodegroup-name\" and (.Value | startswith(\"$nodegroup_prefix\"))))
+           )")
+
+  if [ -z "$asgs" ]; then
+    echo "No Auto Scaling Groups found for EKS cluster '$EKS_CLUSTER_NAME' with nodegroup starting with '$nodegroup_prefix'." >&2
+    return
+  fi
+
+  # Use a for loop to process each ASG
+  for asg in $asgs; do
+    asg_name=$(echo "$asg" | jq -r '.AutoScalingGroupName')
+    original_min_size=$(echo "$asg" | jq -r '.Tags[] | select(.Key == "panfactum/original-min-size") | .Value')
+    original_desired_capacity=$(echo "$asg" | jq -r '.Tags[] | select(.Key == "panfactum/original-desired-capacity") | .Value')
+
+    if [ -z "$original_min_size" ] || [ -z "$original_desired_capacity" ]; then
+      echo "Failed to retrieve original min-size or desired-capacity tags for $asg_name. Skipping." >&2
+      continue
+    fi
+
+    echo "Restoring $asg_name to original min-size: $original_min_size, desired-capacity: $original_desired_capacity" >&2
+
+    # Update the ASG with the original values
+    if aws autoscaling update-auto-scaling-group --auto-scaling-group-name "$asg_name" --min-size "$original_min_size" --desired-capacity "$original_desired_capacity"; then
```

Comment:
Don't update the ASGs directly. You must go through the EKS node group API in order to not cause a conflict.